pkg/upgrade/upgrade_requestor.go

```diff
@@ -306,6 +323,103 @@ func (m *RequestorNodeStateManagerImpl) ProcessUpgradeRequiredNodes(
 
 	return nil
 }
+func (m *RequestorNodeStateManagerImpl) createOrUpdateNodeMaintenance(ctx context.Context,
+	nodeState *NodeUpgradeState) error {
+	// check for existing nodeMaintenance obj and if default prefix is used
+	if nodeState.NodeMaintenance != nil && m.opts.NodeMaintenanceNamePrefix == DefaultNodeMaintenanceNamePrefix {
+		// if exists append requestorID to spec.AdditionalRequestors list
+		nm, ok := nodeState.NodeMaintenance.(*maintenancev1alpha1.NodeMaintenance)
+		if !ok {
+			return fmt.Errorf("failed to cast object to NodeMaintenance. %v", nm)
+		}
+		// check if object is owned by the requestor, if so skip re-creation
+		if nm.Spec.RequestorID == m.opts.MaintenanceOPRequestorID {
+			m.Log.V(consts.LogLevelInfo).Info("nodeMaintenance already exists", nm.Name, "skip creation")
+			return nil
+		}
+
+		// check if requestor is already in AdditionalRequestors
+		if slices.Contains(nm.Spec.AdditionalRequestors, m.opts.MaintenanceOPRequestorID) {
+			m.Log.V(consts.LogLevelInfo).Info("requestor already in AdditionalRequestors list",
+				"requestorID", m.opts.MaintenanceOPRequestorID)
+			return nil
+		}
+
+		m.Log.V(consts.LogLevelInfo).Info("appending new requestor", nm.Spec.RequestorID, "under 'AdditionalRequestors' list")
```

Comment:
nit: restructure this and use `m.opts.MaintenanceOPRequestorID`

m.Log.V(consts.LogLevelInfo).Info("appending new requestor under AdditionalRequestors", "requestor", m.opts.MaintenanceOPRequestorID, "nodeMaintenance", client.ObjectKeyFromObject(nm))